django/test/testcases.py

```diff
@@ -524,17 +524,26 @@ def assertContains(
             response, text, status_code, msg_prefix, html
         )
 
+        content = (
+            b"".join(response.streaming_content)
+            if response.streaming
+            else response.content
+        )
+
         if count is not None:
             self.assertEqual(
                 real_count,
                 count,
                 msg_prefix
-                + "Found %d instances of %s in response (expected %d)"
-                % (real_count, text_repr, count),
+                + "Found %d instances of %s (expected %d) in the following response\n%s"
+                % (real_count, text_repr, count, safe_repr(content)),
             )
         else:
             self.assertTrue(
-                real_count != 0, msg_prefix + "Couldn't find %s in response" % text_repr
+                real_count != 0,
```

Comment:
Unrelated to the change, but this would be better using `self.assertNotEqual()` 🤔 (Can be done as a follow up fixing other similar cases.)